.github/workflows/build_and_test_dbt.yaml

```diff
@@ -0,0 +1,122 @@
+name: Build and test dbt
+
+on:
+  pull_request:
+    branches: [master, data-catalog]
```

Comment:
There are a number of references to `data-catalog` here that cover for the fact that we don't expect this workflow to be pulled into `master` immediately, and that we want to treat `data-catalog` as a CI environment until we're ready to merge `data-catalog` into `master` (https://github.com/ccao-data/data-architecture/issues/55). Part of that merge issue will be stripping out the references to `data-catalog` in these workflows and ensuring that they are ready to be run against `master`.